libcontainer/configs/config.go

```diff
@@ -364,6 +370,17 @@ func (c Command) Run(s *specs.State) error {
 		Stdout: &stdout,
 		Stderr: &stderr,
 	}
+	if s.SeccompFd != 0 {
+		cmd.ExtraFiles = []*os.File{os.NewFile(uintptr(s.SeccompFd), "seccomp-fd")}
```

Comment:
There is something I don't fully understand about `SeccompFd`. It is the file descriptor of the seccomp filter in this `runc` process and it's passed with this number using the container state to the hook. However, according to the [`ExtraFiles` documentation](https://golang.org/pkg/os/exec/#Cmd) it's always 3 in the hook process. 

In my opinion the new `seccompFd` field on state is useless. We can assume that it's always 3 on the sendSeccompFd hooks and document it. It can complicate the implementation a bit, as state is also being used to pass this file descriptor from process_linux.go to `Run()` but we can look for an alternative, like having an specialize type for send seccomp hooks that has `RunWithSeccompFd` or similar call.
What do you think?

Reply:
Yes, it is always going to be 3 with the runc implementation in Golang. I think I preferred to be explicit because the hook could receive several file descriptors if we implement the idea to pass the pidfd too. Then I thought it is nice to have the possibility to have different implementations (runc, crun, etc.) that might or might not pass the same set of fds.

Other software that pass fds via process inheritance also uses an explicit mechanism: see manual page tc-bpf(8): `tc exec` starts a shell with fds specified as environment variables (`$BPF_MAP0`, `$BPF_MAP1` etc.) instead of hardcoding 3.

OTOH, it's not nice to overload the `State` struct with unrelated fields... I was pondering either adding a `addendum` field to make it clearer that it does not really belong to the state of the container, or pass a specialised struct to the SendSeccompFd hook that would include the `State` as a field:

Proposal with addendum:
```
{
    "ociVersion": "0.2.0",
    "id": "oci-container1",
    "status": "running",
    "pid": 4422,    # this is the pid 1 of the container
    "bundle": "/containers/redis",
    "annotations": {
        "myKey": "myValue"
    },
    “addendum”: {
        “seccomp”: {
            "fds": {
                seccomp-fd: 3,
                pidfd: 4,
                memfd: 5
            },
            pid: 4499    # this is the process on which the seccomp policy is attached to (!=pid1)
    }
}
```

Proposal with state as a field:
```
 {
    "seccomp-fd": 3,
    "seccomp-pidfd": 4,
    "seccomp-pid": 4499,    # this is the process on which the seccomp policy is attached to (!=pid1)
    "state": {
        "ociVersion": "0.2.0",
        "id": "oci-container1",
        "status": "running",
        "pid": 4422,    # this is the pid 1 of the container
        "bundle": "/containers/redis",
        "annotations": {
            "myKey": "myValue"
        }
    }
}
```

Wdyt?

Reply:
> Yes, it is always going to be 3 with the runc implementation in Golang. I think I preferred to be explicit because the hook could receive several file descriptors if we implement the idea to pass the pidfd too. Then I thought it is nice to have the possibility to have different implementations (runc, crun, etc.) that might or might not pass the same set of fds.

Oh yes, you're right. Other implementations could not have this value hardcoded and could need this parameter to know what file descriptor it is. 
 
> Wdyt?

I'd go with the second option. It's more clear and explicitly.